Sources/ReactiveArray.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+import ReactiveSwift
+import enum Result.NoError
+
+public enum Change<T> {
+    case insert(element: T, at: Int)
+    case remove(element: T, at: Int)
+}
+
+public final class ReactiveArray<Element> {
+
+    fileprivate var elements: ContiguousArray<Element>
+
+    fileprivate let innerObserver: Observer<[Change<Element>], NoError>
+
+    public let signal: Signal<[Change<Element>], NoError>
+
+    public var capacity: Int {
+        return elements.capacity
+    }
+
+    public init(_ elements: [Element]) {
+        self.elements = ContiguousArray(elements)
+
+        (signal, innerObserver) = Signal<[Change<Element>], NoError>.pipe()
+    }
+
+    deinit {
+        innerObserver.sendCompleted()
+    }
+
+}
+
+extension ReactiveArray: ExpressibleByArrayLiteral {
+
+    public convenience init(arrayLiteral elements: Element...) {
+        self.init(elements)
+    }
+}
+
+extension ReactiveArray: MutableCollection {
+
+    public typealias SubSequence = ArraySlice<Element>
+    public typealias Index = Int
+
+    public var startIndex: Index {
+        return elements.startIndex
+    }
+
+    public var endIndex: Index {
+        return elements.endIndex
+    }
+
+    public subscript(position: Int) -> Element {
+        get {
+            return elements[position]
+        }
+        set {
+            replaceSubrange(position..<position, with: CollectionOfOne(newValue))
+        }
+    }
+
+    public subscript(bounds: Range<Int>) -> SubSequence {
```

Comment:
It could be related to the order of conformances. Try to conform to `RandomAccessCollection` first, then `MutableCollection` and finally `RangeReplaceableCollection`.